src/Workspaces/Remote/ServiceHub/Services/DesignerAttribute/RemoteDesignerAttributeIncrementalAnalyzer.cs

```diff
@@ -0,0 +1,181 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.DesignerAttribute;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.SolutionCrawler;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Remote
+{
+    internal class RemoteDesignerAttributeIncrementalAnalyzer : IncrementalAnalyzerBase
+    {
+        private const string DataKey = "DesignerAttributeData";
+
+        private readonly Workspace _workspace;
+
+        /// <summary>
+        /// Channel back to VS to inform it of the designer attributes we discover.
+        /// </summary>
+        private readonly RemoteEndPoint _endPoint;
+
+        /// <summary>
+        /// Storage where we can keep track of what we know about the project and have informed VS
+        /// (and the project systems) about.
+        /// </summary>
+        private readonly IPersistentStorage _storage;
+
+        public RemoteDesignerAttributeIncrementalAnalyzer(Workspace workspace, RemoteEndPoint endPoint)
+        {
+            _workspace = workspace;
+            _endPoint = endPoint;
+
+            var storageService = _workspace.Services.GetRequiredService<IPersistentStorageService>();
+            _storage = storageService.GetStorage(workspace.CurrentSolution);
```

Comment:
>_storage = storageService.GetStorage(workspace.CurrentSolution); [](start = 12, length = 64)

Assuming the storage needs the solution path? What happens when the solution path changes?

Reply:
> Assuming the storage needs the solution path? What happens when the solution path changes?

Current design for storage (For entire IDE) is that we will continue to use the existing storage location designated for the solution's original location. On next launch of VS we will use a new storage location.  As persistence is purely for caching, and solutions rarely change, this alows us to keep things simple while hitting this 99.99% case easily.

In the extremely rare case, all that happens is that we store a little extra data until next vs launch.

Reply:
Note: it's unclear to me *if* the solution can even change paths while VS is running.  I don't know of any mechanism for this.

Reply:
What if a solution is closed and another is opened?

---
In reply to: [392389608](https://github.com/dotnet/roslyn/pull/42346#discussion_r392389608) [](ancestors = 392389608)